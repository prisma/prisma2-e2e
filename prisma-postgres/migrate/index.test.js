const { PrismaClient, Prisma } = require('@prisma/client')
const fs = require('fs')
const prisma = new PrismaClient()

const pjson = require('./package.json')

describe('tests for prisma postgres', () => {
  afterAll(() => {
    prisma.$disconnect()
  })

  it('should test the Prisma version', async () => {
    expect(Prisma.prismaVersion.client).toEqual(pjson['dependencies']['@prisma/client'])
  })

  it('should query the database', async () => {
    await prisma.user.createMany({
      data: [
        { email: 'alice@prisma.io', name: 'Alice' },
        { email: 'bob@prisma.io', name: 'Bob' },
      ],
    })

    const data = await prisma.user.findMany({
      where: { email: 'alice@prisma.io' },
      select: { email: true, name: true },
    })
    expect(data).toMatchSnapshot()
  })
})
