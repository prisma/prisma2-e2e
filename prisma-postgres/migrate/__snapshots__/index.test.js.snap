// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for prisma postgres should query the database 1`] = `
[
  {
    "email": "alice@prisma.io",
    "name": "Alice",
  },
]
`;
